### PR TITLE
Remove ProxyHelloSignature

### LINKED ISSUE #29610: Remove ProxyHelloSignature
`ProxyHelloSignature` is a custom SSH handshake signature that was used to propagate client IP. It was replaced by signed PROXY headers, but left in the codebase for compatibility with older versions. We can remove it for Teleport v14.

### PR DESCRIPTION
**Pull request by AntonAM**
Removes `ProxyHelloSignature` from the codebase.

Closes #29610

### CODE REVIEW

**Review by rosstimothy — APPROVED** [review]
`+7 −333` 🎉

File: api/observability/tracing/ssh/ssh.go [hunk 1]
```diff
@@ -147,7 +143,7 @@ func Dial(ctx context.Context, network, addr string, config *ssh.ClientConfig, o
 // properly over the ssh connection.
 func NewClientConn(ctx context.Context, conn net.Conn, addr string, config *ssh.ClientConfig, opts ...tracing.Option) (ssh.Conn, <-chan ssh.NewChannel, <-chan *ssh.Request, error) {
 	tracer := tracing.NewConfig(opts).TracerProvider.Tracer(instrumentationName)
-	ctx, span := tracer.Start(
+	_, span := tracer.Start(
```
Comment (strideynet) [inline api/observability/tracing/ssh/ssh.go:146]: Nit: I'd be tempted to keep this `ctx` shadowing. If someone edits the function in future, they'll almost certainly forgot to add this shadow back in and the trace won't be passed.

File: lib/srv/regular/proxy.go [hunk 2]
```diff
@@ -271,14 +265,11 @@ func (t *proxySubsys) proxyToHost(ctx context.Context, ch ssh.Channel, clientSrc
 
 	signer := agentless.SignerFromSSHCertificate(identity.Certificate, authClient, t.clusterName, identity.TeleportUser)
 	aGetter := t.ctx.StartAgentChannel
-	conn, teleportVersion, err := t.router.DialHost(ctx, clientSrcAddr, clientDstAddr, t.host, t.port, t.clusterName, t.ctx.Identity.AccessChecker, aGetter, signer)
+	conn, _, err := t.router.DialHost(ctx, clientSrcAddr, clientDstAddr, t.host, t.port, t.clusterName, t.ctx.Identity.AccessChecker, aGetter, signer)
```
[the remark this answers is not in the document]
Comment (strideynet) [reply lib/srv/regular/proxy.go:268]: Agreed - it'd be nice to drop this :') It might even simplify the implementation of `DialHost` a bit.

**Review by strideynet — APPROVED** [review]
Great to see this finally done !

File: api/observability/tracing/ssh/ssh.go (hunk 1 above)
Comment (AntonAM) [reply api/observability/tracing/ssh/ssh.go]: Done 👍

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**